book/content/material/material_advanced.md

```diff
@@ -0,0 +1,273 @@
+
+my_model.subdomains = [bulk, left_surf, right_surf]
+```
+
+To illustrate how different species' diffusion properties affect the simulation, we set the same boundary condition for each species (1 on the left boundary, 0 on the right). Now we can run the simulation and look at the concentration profile for each species:
+
+```{code-cell} ipython3
+:tags: [hide-input]
+
+# Boundary conditions
+my_model.boundary_conditions = [
+    F.FixedConcentrationBC(left_surf, value=1, species=protium),
+    F.FixedConcentrationBC(right_surf, value=0, species=protium),
+    F.FixedConcentrationBC(left_surf, value=1, species=deuterium),
+    F.FixedConcentrationBC(right_surf, value=0, species=deuterium),
+    F.FixedConcentrationBC(left_surf, value=1, species=tritium),
+    F.FixedConcentrationBC(right_surf, value=0, species=tritium),
+]
+
+my_model.temperature = 300
+my_model.settings = F.Settings(atol=1e-10, rtol=1e-10, final_time=5)
+
+my_model.settings.stepsize = F.Stepsize(1)
+my_model.initialise()
+my_model.run()
+
+import matplotlib.pyplot as plt
+
+def plot_profile(species, **kwargs):
+    c = species.post_processing_solution.x.array[:]
+    x = species.post_processing_solution.function_space.mesh.geometry.x[:,0]
+    return plt.plot(x, c, **kwargs)
+
+for species in my_model.species:
+    plot_profile(species, label=species.name)
+
+plt.xlabel('Position')
+plt.ylabel('Concentration')
+plt.legend()
+plt.show()
+```
+
+Compare this to the case where each species has the same diffusion properties:
+
+```{code-cell} ipython3
+:tags: [hide-input]
+
+import festim as F
+import numpy as np
+
+my_model = F.HydrogenTransportProblem()
+
+protium = F.Species("H")
+deuterium = F.Species("D")
+tritium = F.Species("T")
+my_model.species = [protium, deuterium, tritium]
+
+my_model.mesh = F.Mesh1D(np.linspace(0, 1, 100))
+
+left_surf = F.SurfaceSubdomain1D(id=1, x=0)
+right_surf = F.SurfaceSubdomain1D(id=2, x=1)
+
+mat = F.Material(
+    D_0={protium: 1.0e-3, deuterium: 1.0e-3, tritium: 1.0e-3},  # example diffusivity values for H, D, T
+    E_D={protium: 0.0, deuterium: 0.0, tritium: 0.0},
+)
+
+# One bulk domain using this material
+bulk = F.VolumeSubdomain1D(id=1, borders=[0, 1], material=mat)
+
+my_model.subdomains = [bulk, left_surf, right_surf]
+# Boundary conditions
+my_model.boundary_conditions = [
+    F.FixedConcentrationBC(left_surf, value=1, species=protium),
+    F.FixedConcentrationBC(right_surf, value=0, species=protium),
+    F.FixedConcentrationBC(left_surf, value=1, species=deuterium),
+    F.FixedConcentrationBC(right_surf, value=0, species=deuterium),
+    F.FixedConcentrationBC(left_surf, value=1, species=tritium),
+    F.FixedConcentrationBC(right_surf, value=0, species=tritium),
+]
+
+my_model.temperature = 300
+my_model.settings = F.Settings(atol=1e-10, rtol=1e-10, final_time=5)
+my_model.settings.stepsize = F.Stepsize(1)
+my_model.initialise()
+my_model.run()
+
+import matplotlib.pyplot as plt
```

Comment:
Same comment as above, split this code cell here and only hide the matplotlib stuff